test/userResDecorator.js

```diff
@@ -51,57 +61,74 @@ describe('userResDecorator', function () {
       }
     }));
 
-    request(app).get('/').end(done);
+    request(app).get('/get').end(done);
   });
 
   it('works with promises', function (done) {
     var app = express();
-    app.use(proxy('https://httpbin.org', {
+    var handler = {
+      method: 'get',
+      path: '/promise',
+      fn: function(req, res) {
+        res.json({ origin: '127.0.0.1' });
+      }
+    };
+    proxyServer.close();
+    proxyServer = proxyTarget(12345, 100, [handler]);
+
+    app.use(proxy('localhost:12345', {
       userResDecorator: function (proxyRes, proxyResData) {
         return new Promise(function (resolve) {
-          proxyResData.funkyMessage = 'oi io oo ii';
+          var data = JSON.parse(proxyResData.toString('utf8'));
+          data.funkyMessage = 'oi io oo ii';
```

Comment:
This is weird. I see that the way we're generating responses is different, but I want to re-read this more closely.